services/firebase/src/lib/digest.ts

```diff
@@ -0,0 +1,130 @@
+const { createSSRApp }  = require('vue');
+const { renderToString } = require('vue/server-renderer');
+import mjml2html = require("mjml")
+import * as fs from "fs"; 
+import { firestore } from "../firebase";
+
+
+export async function renderEmail(type: string, data:any, customUtms = {}) {
+  const template = fs.readFileSync(`./templates/${type}.mjml`, "utf8");
+
+  const defaultUtms = {
+    campaign: type,
+    medium: "email",
+    source: "newsletter",
+  };
+
+  const utm = {
+    ...defaultUtms,
+    ...customUtms,
+  };
+
+  const app = createSSRApp({
+    data: () => {
+      return data;
+    },
+    template,
+    methods: {
+      link(url:string, utmContent = "") {
+        return (
+          url +
+          "?utm_campaign=" +
+          utm.campaign +
+          "&utm_medium=" +
+          utm.medium +
+          "&utm_source=" +
+          utm.source +
+          "&utm_content=" +
+          utmContent
+        );
+      },
+    },
+  });
+
+  app.config.compilerOptions.isCustomElement = (tag:any) => tag.startsWith("mj");
+
+  return  mjml2html(await renderToString(app)).html;
+}
+
+export async function getWeeklyData(city:string) {
+
+  const today = new Date().toISOString().slice(0,10)
+  const then = new Date();
+  then.setDate(then.getDate() + 7);
+  const sevenDaysFromNow = then.toISOString().slice(0,10); 
+  const data = []
+  let cityData: any = {}; 
+  let profile: any =  {};
+
+  const cityDocs = (await firestore
+  .collection('profiles') 
+  .where('username', '==', city)
+  .get()
+  ).docs
+
+  for(const doc of cityDocs ) {
+    const city = {
```

Comment:
```suggestion
   cityData = {
```

you can do it right here, line 71 seems useless. you use city only one time.
also `city` is name of function parameter, it's already taken, you need to use another name